src/sync/index.js

```diff
@@ -75,8 +75,17 @@ async function sync(db) {
   const topicsNeedBalanceUpdate = new Set();
   const oraclesNeedBalanceUpdate = new Set();
 
-  let currentBlockChainHeight = await qclient.getBlockCount();
-  currentBlockChainHeight -= 1;
+  let retr = true;
+  while(retr){
+    try{
+      var currentBlockChainHeight = await qclient.getBlockCount();
+      currentBlockChainHeight = Math.max(0,currentBlockChainHeight-1);
+      retr = false;
+    }
+    catch (exc){
+      retr = true;
+    }
+  }
```

Comment:
this entire logic seems to be a hacky fix for the core problem. the main problem is that `startSync()` is being called before `qtumd` is done initializing right? 

can you try to solve it in `src/index.js`? basically need to wait for `qtumd` to finish initializing before starting the other services.